### PR TITLE
Updated Clarify NLP example notebook

### PR DESCRIPTION
**Pull request by oyangz**
*Issue #, if available:* https://issues.amazon.com/issues/RAI-3992

*Description of changes:*

* Fixed dependency conflicts
* Reduced complexity of code blocks to improve readability
* Updated deadlinks and typos
* Added example for different tokenization granularities

*Testing done:*
Ran the notebook end to end and confirmed there were no errors.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-sagemaker-examples/blob/master/CONTRIBUTING.md) doc and adhered to the example notebook best practices
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-sagemaker-examples/blob/master/README.md)
- [x] I have tested my notebook(s) and ensured it runs end-to-end
- [x] I have linted my notebook(s) and code using `black-nb -l 100 {path}/{notebook-name}.ipynb`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: sagemaker-clarify/text_explainability/text_explainability.ipynb [hunk 1]
```diff
@@ -7,20 +7,30 @@
     "# Explaining text sentiment analysis using SageMaker Clarify"
```
Comment (larroy) [inline sagemaker-clarify/text_explainability/text_explainability.ipynb:7]: <p>maybe you can annotate the types in the function and add a docstring showing what shap_explanations data look like? a "head" from them showing the data structure would help.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3793/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='60'/>

Comment (larroy) [inline sagemaker-clarify/text_explainability/text_explainability.ipynb:7]: <p>I would add types to the function definition</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3793/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='64'/>

Comment (larroy) [inline sagemaker-clarify/text_explainability/text_explainability.ipynb:7]: <p>can we delete the predictor?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3793/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='74'/>

**Comment by larroy** [comment]
Good work, LGTM. Thanks!

**Review by atqy — APPROVED** [review]

**Pull request merged**